Tests/SentryTests/SentryLogBatcherTests.swift

```diff
@@ -66,4 +292,22 @@ class SentryLogBatcherTests: XCTestCase {
             attributes: attributes
         )
     }
+    
+    private func waitBeforeTimeout() {
+        // Wait for timer to fire
+        let expectation = self.expectation(description: "Wait for async add")
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.02) {
+            expectation.fulfill()
+        }
+        waitForExpectations(timeout: 0.5)
+    }
+    
+    private func waitAfterTimeout() {
+        // Wait for timer to fire
+        let expectation = self.expectation(description: "Wait for timer flush")
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.2) {
+            expectation.fulfill()
+        }
+        waitForExpectations(timeout: 0.5)
+    }
```

Comment:
`m`: I think we shouldn't use the actual dispatch queue for most tests but instead we should add a new method to the  dispatchQueueWrapper for `asyncAfter` and the workItem as a parameter. This will speed up the tests and make them more reliable. Still, we could have one or two integration test ensuring the code works correctly with an acutal dispatch queue.